src/bin/sol-fbp-generator/main.c

```diff
@@ -233,8 +232,7 @@ read_meta_type_file(const struct sol_flow_metatype_context *ctx,
     }
 
     contents = sol_file_reader_get_all(reader);
-    *buf = contents.data;
-    *size = contents.len;
+    *buf = SOL_BUFFER_INIT_CONST((char *)contents.data, contents.len);
```

Comment:
@barbieri, Actually, using ```sol_str_slice``` was exactly what I wanted to do in the beginning. I will look into this ```sol_blob``` structure, and see how it can be used in this context. For now, I will take out this commit out of this PR so the other commits could be integrated.